src/main/java/org/elasticsearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -268,6 +269,9 @@ protected void doRun() {
                                     .withType(TransportRequestOptions.Type.RECOVERY)
                                     .withTimeout(recoverySettings.internalActionTimeout());
 
+                            // When we last consulted RateLimiter:
+                            long lastPauseCheckBytes = 0;
```

Comment:
> this needs to be out of the file loop, no? which also means we need a total read count

I thought this was OK?  It just means small files (< 100 KB @ 20 MB/sec) are not counted, nor the final chunk of a big file if it's < 100 KB, right?

But if you think this really matters, I can pull it out / also use AtomicLong, or maybe make the RateLimiterWrapper....